libs/state/src/lib/core/utils/object-diver.ts

```diff
@@ -0,0 +1,10 @@
+import { isKeyOf } from './typing';
+
+export function objectDiver<T>(
+  obj: T | Partial<T>,
+  keys: string[]
+): T | Partial<T> {
+  const key = keys.shift();
+  const value = isKeyOf(key) ? obj[key] : null;
+  return !!value ? (keys.length > 0 ? objectDiver(value, keys) : value) : obj;
+}
```

Comment:
Here I suggest working immutable with the keys array as we would mutate a user variable.
Furthermore, TS should work correctly. Have a look at the pluck operator or similar to get an idea of the typing.
maybe consider a more intuitive naming as e.g. `safePluck`.

```suggestion
export function objectDiver<T>(
  obj: Partial<T>,
  keys: ????[]
): ??? {
  let prop = obj;
  keys.forEach((key) => {
    if(prop[key] !== undefined) {
      prop = prop[key];
    }
  })
  return prop;
}
```

Reply:
https://github.com/BioPhoton/rx-angular/pull/161/commits/cf04a730b730345277c473f302a516bf5697f9b5
https://github.com/BioPhoton/rx-angular/pull/161/commits/c8e03aea2a094c6eaa642aff5e010dad51203e99